### DISCUSSION TITLE
🚨RESOLVED: Network disruption🚨

### DISCUSSION THREAD

**Discussion by redfish31** [codemagic-ci-cd/codemagic-docs#2424] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Try to installing SDK

### Expected results

Download & Install

### Actual results

```
== Use Flutter channel stable ==
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.3-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.3-stable.zip
curl: (28) Operation timed out after 45003 milliseconds with 276924723 out of 1167817719 bytes received

Downloading Flutter release timed out, try again...
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.3-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.3-stable.zip
curl: (28) Operation timed out after 45003 milliseconds with 545429997 out of 1167817719 bytes received

Downloading Flutter release timed out, try again...
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.3-stable.zip --connect-timeout 10.00 --max-time 45.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.3-stable.zip
curl: (28) Operation timed out after 45001 milliseconds with 118802687 out of 1167817719 bytes received

Downloading Flutter release timed out

Build failed :|
Downloading Flutter release timed out
```

**Comment by mikhail-tokarev** [comment]
We are seeing an increased number of network errors when interacting with Google Cloud Storage and our team is looking how we can improve things.

**Reply by briacwg** [reply]
This is Google's issue then?

**Reply by mikhail-tokarev** [reply]
we believe it is. We are experiencing similar issues across 3 different data centres in US, however nothing on https://status.cloud.google.com to prove it.

Currently we are trying to move some files from GCS to AWS S3. Next update when we have more information.

**Reply by mikhail-tokarev** [reply]
Additionally we have increased timeout to download Flutter SDK. I would appreciate it if somebody has builds running and could post the results after applied changes.

**Reply by selawik** [reply]
Still not enough time

**Reply by Saifallak** [reply]
worked with me, too slow, but works at least.

```log

== Parsing flutter version from FVM config ==

== Use Flutter version 3.19.1 ==
> curl --fail --show-error --silent --output flutter_macos_arm64_3.19.1-stable.zip --connect-timeout 10.00 --max-time 120.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.19.1-stable.zip

> 7z -bd x flutter_macos_arm64_3.19.1-stable.zip

7-Zip [64] 17.04 : Copyright (c) 1999-2021 Igor Pavlov : 2017-08-28
p7zip Version 17.04 (locale=utf8,Utf16=on,HugeFiles=on,64 bits,4 CPUs LE)

Scanning the drive for archives:
1 file, 1167172797 bytes (1114 MiB)

Extracting archive: flutter_macos_arm64_3.19.1-stable.zip
--
Path = flutter_macos_arm64_3.19.1-stable.zip
Type = zip
Physical Size = 1167172797

Everything is Ok

Folders: 2795
Files: 10318
Size:       2647303416
Compressed: 1167172797

Flutter 3.19.1 • Dart 3.3.0
```

**Reply by lpickford-pixelbeard** [reply]
Can check our latest build here:
https://codemagic.io/app/64b935368deaa4bc262877d3/build/65fc5e6ce47d9c26b198f711

**Reply by selawik** [reply]
The next try was ok. Thank you!

**Reply by gsi-eric** [reply]
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/24942358/fd1b9776-5f34-4b1f-87b1-727c5a2b5544)
Still not enough time

**Reply by kieranmccwayleadr** [reply]
I wonder would it be useful to allow caching of SDK's? Similar like we do for dependencies. I understand 1.1GB is a large enough folder across multi users. But maybe have a codemagic SDK cache. Once downloaded, all flutter users can pull from it? Speed up the download while also removing the Google dependency? Worth a thought 🤔

**Reply by mikhail-tokarev** [reply]
the problem only that our cache storage also on GCS 🙃 but after this incident there is definitely food for thought.

**Reply by briacwg** [reply]
The rule that I have is never rely on one service, always have a backup plan.

**Reply by elemental21** [reply]
Finally got it after maybe 10 builds!

**Reply by AnonymousAliensX** [reply]
When will this issue be resolved, and how much time might it take? I have tried many time, still getting time out error !

**Reply by gsi-eric** [reply]
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/24942358/e45e28b4-c06c-44b0-8995-bfe3aaa7412f)
Finally got it

**Reply by mikhail-tokarev** [reply]
We are removing load from Google Cloud Storage where possible and can see that the number of errors is slowly decreasing. We are continuing to monitor the situation.

**Reply by mikhail-tokarev** [reply]
Error rates have stabilized. We continue to monitor the situation closely.

**Reply by mikhail-tokarev** [reply]
We will provide a comprehensive incident report later next week. However, here is a brief summary:
1. The incident only affected the Atlanta data center and does not appear to be related to Google Cloud Storage. However, the majority of our macOS machines are located in Atlanta.
2. The confusion came from a combination of misleading information from our internal monitoring system and the absence of alerts from the Atlanta data center. We will collaborate with our data center partner to better understand the issue.
3. Currently, the situation has stabilized due to a natural decrease in network load and the implementation of various fixes. Additionally, we are working on internal optimizations to significantly reduce network load and aim to release these improvements soon.
4. I sincerely apologize for the extended disruption and want to assure you that we are committed to investing our time to improve the stability and reliability of our service. Thank you for your patience!

**Reply by kieranmccwayleadr** [reply]
Thanks for the update, but it was just timeouts, no panic 🥇 . Just shows everyone uses Codemagic 👍

**Reply by lpickford-pixelbeard** [reply]
@mikhail-tokarev Is this still an issue? We're experiencing it still on our builds for one of our projects, here's a link to the latest:
https://codemagic.io/app/64b935368deaa4bc262877d3/build/65fd939a779c853d2d1d8d84

**Reply by helinanever** [reply]
Unfortunately, there have been new incidents today, causing both timeouts and Something Went Wrong errors. Please see the status page here: https://status.codemagic.io/

**Reply by ggirotto** [reply]
@helinanever the issue is marked as resolved in the status page but the issues are still happening

**Reply by Imgkl** [reply]
Yes. We are facing this issue.

**Reply by peer-f** [reply]
Facing this issue too. It worked fine in the morning but now it fails every time.

**Reply by mrruby** [reply]
It's still the issue for me

**Comment by markbreuss** [comment]
We are still seeing timed out builds.

Build ID:
66019467d09bc98f417de6f6

Screenshot:
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/69714845/dfa500d7-1693-4c77-981a-9a75805b3a52)

**Reply by ChrisElliotUK** [reply]
Same for us. 

Build ID: 
660195155f9460b428104153

**Comment by mikhail-tokarev** [comment]
Unfortunately the network performance issue has returned. We are working with our data center partner to resolve the issue.  Please see the status page here: https://status.codemagic.io/

Additionally we are trying to provide more machines in other data centers to reduce the impact of the issue.

Thank you for your patience!

**Reply by mikhail-tokarev** [reply]
we are still waiting for the update from our data center partners :(((

**Comment by ChrisElliotUK** [comment]
Annoyingly this issue seems to be back again this morning. We managed two builds before getting a timeout. 
build id: 6602945540a27391b82e2b84

**Reply by YannMarti** [reply]
same for us, builds have been failing the past 30 minutes.

**Reply by lstimac** [reply]
This issue has been occurring every work day since Thursday, it seems that it is not being resolved - it just gets better during the hours when not that many people are working which is probably why it worked during the weekend. I also started seeing queued builds, before this started they would always run immediately :/

Will this issue get resolved permanently?

**Reply by helinanever** [reply]
We can confirm that the issue still exists and becomes more severe as the load on our infrastructure increases. Our data center partner is still investigating the problem but unfortunately, there is no progress so far. We apologize for the inconvenience caused!

**Reply by sagang-olimind** [reply]
We have a major demo this morning and code magic keeps failing on this step. keeping a close eye and hoping we can get this cleared out. going manually through app store connect and pushing the app out

**Reply by mikhail-tokarev** [reply]
Hi @sagang-olimind 
sorry to hear that. Can you use Linux for demo?

**Reply by lstimac** [reply]
@helinanever Could you update https://status.codemagic.io/? It does not reflect current status and we could then also get the notification once the issue is resolved :)

**Reply by sagang-olimind** [reply]
@mikhail-tokarev  - its an iOS demo

**Reply by EclipseAndrey** [reply]
We have same problem now

**Reply by EclipseAndrey** [reply]
Coincidentally, I have a critical bug in production and I can't publish the release, since it's about 15 applications with different settings. We are really looking forward to solving the problem!

**Reply by cgn170** [reply]
We have the same problem too

**Reply by aclowkey** [reply]
Yup, since 9:30Z can't build my iOS app. Status page seems green tho.

**Reply by mikhail-tokarev** [reply]
sorry, indeed overlooked status page update.

**Reply by HussainTaj-W** [reply]
I hope this gets resolved soon. I don't even remember the steps or have the setup to build my app locally. I exclusively use CodeMagic for my builds.

**Reply by kieranmccwayleadr** [reply]
Codemagic knows it's an issue but are stuck with GCS to fix, it's not CM

For those who want to build and deploy use the docs:

https://docs.flutter.dev/deployment/android
https://docs.flutter.dev/deployment/ios

I'm sure it'll die down soon, but don't wait until it's fixed. Look at the build logs that the CM builds do. It'll be like running locally also.

**Reply by Yogesh-Dubey-Ayesavi** [reply]
Same here

**Reply by jkoenig134** [reply]
Same here.

**Reply by lstimacay** [reply]
@kieranmccwayleadr We are literally in the process of migration from Github Actions to Codemagic (Pay as you go tier) so this is a bit discouraging 😅

**Reply by kieranmccwayleadr** [reply]
@lstimacay ah totally get you. Look we have done loads of builds with CM, this is the first time this has come up. It's not great, and sometimes the builds do get through. I have gotten say 1/4 builds passed the time out. But at the end of the day it's a super tool, that is getting bottleneck issues through Google (of all people :/). 

While discouraging, we haven't had this issue before so it's still a great migration. But definitely don't be self blocking if you are still able to deploy to the app stores, they aren't down 👍

**Reply by Yogesh-Dubey-Ayesavi** [reply]
This happens very rarely

**Reply by lstimacay** [reply]
@kieranmccwayleadr I was really impressed when I played around with it a couple of months back 👌 I think that timing is unfortunate because I'm currently in the process of preparing all of the workflows before the migration so I'm constantly testing and running builds 😅 

Fingers crossed this gets resolved soon 🙏

**Reply by kieranmccwayleadr** [reply]
@lstimacay completely xD. Also look into the Firebase Distribution hook they have under the Distribution section. It's such a great tool if you use Firebase and works "Magic" with workflow 😆

**Reply by elianortega** [reply]
Same issue here, I have attempted 6 builds, none of them succeeding

**Reply by opanitch** [reply]
@Yogesh-Dubey-Ayesavi 3rd separate incident (different reasons) that has completely frozen development for us in as many months.

**Reply by AlainMk** [reply]
The issue is still there. Can’t deploy anymore.

**Reply by drpark** [reply]
same here

**Reply by warpaltarpers** [reply]
I think if we keep telling Codemagic that it's not working, the off-site data center will get fixed faster 🤔

**Reply by mikhail-tokarev** [reply]
thank you for your understanding! 
However right now we are receiving confusing information and I need your help 🙏 Can somebody please share build id failed with network issue after 18:30UTC?

**Reply by lstimacay** [reply]
@mikhail-tokarev Here it is: 66031729b11f45d5560ed4e3

Failed with:

```
== Use Flutter version 3.3.9 ==
> curl --fail --show-error --silent --output flutter_macos_arm64_3.3.9-stable.zip --connect-timeout 10.00 --max-time 120.00 https://storage.googleapis.com/flutter_infra_release/releases/stable/macos/flutter_macos_arm64_3.3.9-stable.zip
curl: (28) Operation timed out after 120006 milliseconds with 514787713 out of 1007490932 bytes received

Downloading Flutter release timed out

Build failed :|
Downloading Flutter release timed out
```

**Reply by warpaltarpers** [reply]
If it's at all helpful for triage, we have had 3 builds succeed: 

- **Started at 19:01UTC:** 66031b8263c492b39d7b552d
- **Started at 19:17UTC:** 66031d1328e00f2f74a2e10e
- **Started at 19:28UTC:** 660321c3538a95a262e29dd5

**Comment by burekas7** [comment]
So what should we need to do here? Just waiting and to try later again?

**Comment by tony123S** [comment]
just tried, still not working...

**Comment by lucas-sesti** [comment]
Just tried now, still not working too..

**Comment by ggirotto** [comment]
Please stop spamming the discussion. Commenting "me too", "not working" or anything related WILL NOT fix the issue. Codemagic team is aware of the issue and they'll post any update once they have it.

**Comment by burekas7** [comment]
> Unfortunately the network performance issue has returned. We are working with our data center partner to resolve the issue. Please see the status page here: https://status.codemagic.io/
> 
> Additionally we are trying to provide more machines in other data centers to reduce the impact of the issue.
> 
> Thank you for your patience!

You can check for updates here:
https://status.codemagic.io/

**Comment by salimazeriDev** [comment]
Same here, we had that problems on Friday, and today we have same situation.

**Comment by mrruby** [comment]
looks like its working for me now

**Reply by salimazeriDev** [reply]
Yeah, I see that my builds passed that failing step. Hope, it's not just a luck as I have few builds to make today.

**Reply by emanuel-braz** [reply]
not for me 😔

**Reply by burekas7** [reply]
Now it's working for me too.
But the whole build process takes much longer then usual.

**Comment by sobe-wt** [comment]
Happening for me now ;(

**Reply by warpaltarpers** [reply]
Seems like, as stated by @mikhail-tokarev before, the increased load during peak work hours can bring the issue out

**Comment by mikhail-tokarev** [comment]
First of all, I’m sorry for the long silence in this thread. I'd like to provide an update on the current status and our next steps to resolve the problem. I want to assure all of you that we are taking the situation very seriously and doing everything possible to address it. I'll be happy to answer any questions you may have in the comments.

As you are aware, we are still experiencing significant challenges with network performance. Currently, the data center has identified issues with the hardware of the firewalls, caused by high traffic. These issues are causing a high packet drop ratio, resulting in slow download speeds and timeouts.

- We did not receive any notifications of the traffic increase beforehand, leaving us unprepared to address the issue proactively.
- Furthermore, we are unable to quickly upgrade the firewalls or add more machines to another data center, because our partner has encountered issues with the supply chain.

> We weren’t prepared for such a scenario, and we completely understand that we have performed poorly, putting you in a situation where you cannot rely on our service anymore.

Currently, we are planning the following steps to mitigate the issue:
1. As a workaround, the data center will setup backup firewalls to create a new subnet to reduce the firewall load (**UPDATE: Maintenance `Mar 28 at 02:00pm UTC  –  Mar 28 at 05:00pm UTC`**).
2. We expect the first machines to arrive later this week to help distribute our load.
3. We are also exploring alternative solutions to prevent such incidents in the future.

I sincerely appologize for the troubles caused!

**Reply by thassio-vinicius** [reply]
Are the workarounds already in place? If not, any ETA? I tried making a new build just now and it's still failing

**Reply by mikhail-tokarev** [reply]
unfortunately, not yet. I'll update the thread once we know anything from the data center engineers.

**Reply by mikhail-tokarev** [reply]
At this moment, we still haven't received any information or confirmation from the data center engineers. We expect emergency maintenance tomorrow. I apologize for yet another delay.

**Comment by Imgkl** [comment]
Thanks @mikhail-tokarev for the update and the action plan. 

FWIW the builds are in queue for the last 30 mins.

**Reply by mikhail-tokarev** [reply]
thanks, we are currently restoring machines to the pool as they lost network connection (almost simultaneously). We haven't received any announcement and not sure if the data center is making any changes or not. However all builds should start soon.

**Comment by helinanever** [comment, marked as answer]
There is going to be **emergency maintenance starting at 14:00 UTC today** with an estimated duration of 3 hours.

Our data center partner will move the Apple M1 machines (`mac_mini_m1`) to a new network to improve network performance.

During the maintenance window, we will be operating with a reduced number of M1 build machines. As a result, you may be experiencing longer than usual build queues.

Apple M2 and Mac Pro, Linux and Windows machine availability will not be affected.

Thank you for your understanding.

Please keep an eye on the status page for updates: https://status.codemagic.io/

**Reply by mikhail-tokarev** [reply]
The planned work takes longer than expected, currently we increased the maintenance window for another 3 hours.

**Reply by mikhail-tokarev** [reply]
The data center engineers have completed the new subnet configuration. We are now awaiting an update on the ETA for when the machines will be physically switched to the new network.

**Reply by mikhail-tokarev** [reply]
The data center wasn't able to start the second part of the planned maintenance, and we expect another emergency maintenance tomorrow.

To resolve the build queue backlog, we will temporarily route all builds to Apple M2 hosts, which have the same software versions but are faster. This solution will not result in any additional costs.

**Reply by erik-wayhaven** [reply]
The status page says "All services are online" but I have had builds stuck in queuing for the last few hours and still can't get one through. Is there still an ongoing issue?

**Reply by mikhail-tokarev** [reply]
It still takes quite some time to resolve the build queue backlog.

**Reply by mikhail-tokarev** [reply]
Currently, all services are operating as expected. We continue to monitor the situation closely and are working with data center engineers to improve network stability.

**Reply by mikhail-tokarev** [reply]
There is another ongoing emergency maintenance, so we will be operating with a reduced number of M1 build machines. As a result, you may experience longer than usual build queues.

Please keep an eye on the status page for updates: https://status.codemagic.io/

**Reply by mikhail-tokarev** [reply]
We need to extend the maintenance period. We apologize for any inconvenience this may cause.

**Reply by mikhail-tokarev** [reply]
The maintenance has been completed and all builds will start soon.

**Comment by ChrisElliotUK** [comment]
Thanks Mikhail. I really appreciate the update and the updates on GitHub. The communication has been very good so massive applause to your team. Christopher ElliotOn 28 Mar 2024, at 19:14, Mikhail Tokarev ***@***.***> wrote:﻿
The data center engineers have completed the new subnet configuration. We are now awaiting an update on the ETA for when the machines will be physically switched to the new network.

**Comment by irby** [comment]
Thank you for your hard work on this, guys!

**Comment by bastienlemaitre** [comment]
Any update ?

**Reply by mikhail-tokarev** [reply]
sorry, just posted update here https://github.com/orgs/codemagic-ci-cd/discussions/2424#discussioncomment-8940664

**Comment by eduardoazvd17** [comment]
Its still not working for me, im having timeout issues on installing dependencies section. I haven't faced it before.

![Screenshot 2024-03-29 at 12 02 49](https://github.com/codemagic-ci-cd/codemagic-docs/assets/49172682/d25bcce0-6317-4630-a92d-c6aa3c6deceb)

**Reply by mikhail-tokarev** [reply]
can you share the build id?

**Reply by eduardoazvd17** [reply]
Of course, ID: 6606c5f9f13b41ec5f927dff

**Reply by eduardoazvd17** [reply]
I tryed start a new build but its keep queued: 6606d73cde39e009d29e17cf

**Reply by mikhail-tokarev** [reply]
thanks for sharing build id. I'll try to check time issue with time out
re queued builds I've post update here https://github.com/orgs/codemagic-ci-cd/discussions/2424#discussioncomment-8953207

**Reply by mikhail-tokarev** [reply]
The machine was disconnected from the network while running your build `6606c5f9f13b41ec5f927dff`. Currently, we are awaiting any confirmation from the data center engineers. However, it seems they started reconfiguring hosts without letting us know, so we didn't have a chance to remove hosts from the pool :(

**Reply by eduardoazvd17** [reply]
Ok, thanks for reply. My build was completed successfully now.

**Comment by momer** [comment]
I'm also encountering a stuck, queued, build at build id: `6606dada350295535e82475b`

**Reply by mikhail-tokarev** [reply]
check update here https://github.com/orgs/codemagic-ci-cd/discussions/2424#discussioncomment-8953207

**Comment by jhonny218** [comment]
My build (6606c9400f14a76b72028590) took an hour and never finished. I had to cancel it.
Started a new build (6606da07de39e009d29e18db) has been queued for more than 30 mins now

**Reply by mikhail-tokarev** [reply]
I've posted update here https://github.com/orgs/codemagic-ci-cd/discussions/2424#discussioncomment-8953207

**Comment by mikhail-tokarev** [comment]
🔔 Important Announcement 🔔

We sincerely apologize for the recent network issues that have impacted many of you. We understand the frustration that comes with being unable to use CI/CD and release new versions, and we want to assure you that ensuring service reliability is our top priority.

As a gesture of our apology and appreciation for your patience during this time, we're offering you a 50% discount on your next payment. Simply use the code **`WEARESORRY`** in your team billing settings to redeem your discount.

We are committed to providing you with the best possible experience. If you have any further questions or concerns, please don't hesitate to reach out to me or our customer support team.

**Reply by ggirotto** [reply]
@mikhail-tokarev How to get this discount for those in annual billing plans?

**Reply by mikhail-tokarev** [reply]
Hey @ggirotto ! please ask the support team using in-app chat widget

**Reply by dtrdic** [reply]
Just a note for users who don't know how to apply the coupon. The promotion code can be applied in the billing team settings under the billing overview section.

![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/dd175ee6-7244-49c4-a9c5-e7dacf02980e)

**Comment by himesh-cm** [comment]
While we are closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!

**Comment by matmicro** [comment]
Facing same issue currently.
Any ETA to get this resolved ?

**Reply by mikhail-tokarev** [reply]
hey, this issue has beed resolved long time ago. You can try to restart your build and if issue persists, open a new thread, and provide build id and more details please. Someone from the team will take a look. Thank you!
